src/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -328,6 +328,22 @@ public void Match_Timeout()
             Assert.Equal("a", match.Value);
         }
 
+        [Fact]
```

Comment:
tests for hitting each of the exceptions?

Reply:
Originally I wanted to keep the change as small as possible because of different channels we wanted to port it. But now that doesn't matter anymore and I will add more test coverage.